code/modules/vending/_vending.dm

```diff
@@ -1194,3 +1195,48 @@ GLOBAL_LIST_EMPTY(vending_products)
 		var/obj/item/I = target
 		I.custom_price = price
 		to_chat(user, "<span class='notice'>You set the price of [I] to [price] cr.</span>")
+/**
+ * Vend the specified item
+ *
+ * Arguments:
+ * * product - The item that should be vended
+ */
+/obj/machinery/vending/proc/vend_item(var/datum/data/vending_product/product)
+	var/obj/item/vended_item = new product.product_path()
+	product.amount--
+	if (prob(3))
+		//Item got stuck in vending machine
+		LAZYADD(stuck_items, vended_item)
+		to_chat(usr, "<span class='warning'>It seems like [product.name] got stuck in the vending machine!<br>\
+		Maybe you can get it unstuck by buying another [product.name] or by bashing the vending machine.</span>")
```

Comment:
This is a bit too much, you would think they could figure out how to get it unstuck by themselves. Just "[product.name] gets stuck in the vending machine!" would suffice

Reply:
I guess people could figure out to just bash the vending machine, but im not sure if they would realize that ordering the same item can also get it unstuck? Maybe they could think all types of that item are now stuck and not order it anymore?
I also want to really make sure that people know they can just punch the vendor and get the item very likely back.